packages/discord.js/typings/index.d.ts

```diff
@@ -2640,6 +2640,8 @@ export class ThreadChannel<Forum extends boolean = boolean> extends TextBasedCha
   public setLocked(locked?: boolean, reason?: string): Promise<AnyThreadChannel>;
   public setName(name: string, reason?: string): Promise<AnyThreadChannel>;
   public setAppliedTags(appliedTags: Snowflake[], reason?: string): Promise<ThreadChannel<true>>;
+  public pin(reason?: string): Promise<ThreadChannel<true>>;
+  public unpin(reason?: string): Promise<ThreadChannel<true>>;
```

Comment:
Since those methods return `this.edit()`, which is has a return type of `Promise<AnyThreadChannel>` (the other methods with the exception of `setAppliedTags`, for some reason, also follow this), the two new methods should also follow for consistency:
```suggestion
  public pin(reason?: string): Promise<AnyThreadChannel>;
  public unpin(reason?: string): Promise<AnyThreadChannel>;
```